lms/djangoapps/course_api/fields.py

```diff
@@ -0,0 +1,26 @@
+""" Custom fields for Django REST Framework. """
+
+from __future__ import unicode_literals
+
+from rest_framework import fields
+
+
+class AbsoluteURLField(fields.URLField):
+    """
+    Field that serializes values to absolute URLs based on the current request.
+
+    If the value to be serialized is already a URL, that value will returned.
+    """
+
+    def to_representation(self, value):
+        request = self.context.get('request', None)
+
+        assert request is not None, (
```

Comment:
Fair enough. I'm just not a big fan of having client code catch `AssertionError`, since we wouldn't want to subclass it if in the future we decide to have more granular exception information.